File: frontend/src/types/history.ts
/** TypeScript types for comparison history API. */

export interface HistoryRecord {
  id: number;
  task_id: number | null;
  source_connection_id: number;
  target_connection_id: number;
  source_table: string;
  target_table: string;
  compare_mode: string;
  source_row_count: number | null;
  target_row_count: number | null;
  diff_count: number;
  diff_percentage: number | null;
  has_critical_diffs: boolean;
  started_at: string;
  completed_at: string | null;
  status: string;
  error_message: string | null;
  result_summary: Record<string, unknown> | null;
  created_at: string;
}

export interface TrendDataPoint {
  date: string;
  diff_count: number;
  completed_count: number;
}

export interface TrendResponse {
  period: 'daily' | 'weekly' | 'monthly';
  data_points: TrendDataPoint[];
  total_comparisons: number;
  total_diffs: number;
  avg_diff_count: number;
}

export interface HistoryStats {
  total_comparisons: number;
  completed: number;
  failed: number;
  avg_diff_count: number;
  max_diff_count: number;
  last_24h_comparisons: number;
  last_7d_comparisons: number;
}

File: backend/app/schemas/history.py
"""Pydantic schemas for comparison history API."""

from pydantic import BaseModel, Field
from typing import Optional, List, Dict, Any
from datetime import datetime


class HistoryRecord(BaseModel):
    """Single comparison history record."""
    id: int
    task_id: Optional[int]
    source_connection_id: int
    target_connection_id: int
    source_table: str
    target_table: str
    compare_mode: str
    source_row_count: Optional[int]
    target_row_count: Optional[int]
    diff_count: int
    diff_percentage: Optional[float]
    has_critical_diffs: bool
    started_at: datetime
    completed_at: Optional[datetime]
    status: str
    error_message: Optional[str]
    result_summary: Optional[Dict[str, Any]]
    created_at: datetime

    class Config:
        from_attributes = True


class TrendDataPoint(BaseModel):
    """Data point for trend chart."""
    date: str
    diff_count: int
    completed_count: int


class TrendResponse(BaseModel):
    """Response schema for trend analysis."""
    period: str  # daily, weekly, monthly
    data_points: List[TrendDataPoint]
    total_comparisons: int
    total_diffs: int
    avg_diff_count: float


class HistoryStats(BaseModel):
    """Statistics summary for comparison history."""
    total_comparisons: int
    completed: int
    failed: int
    avg_diff_count: float
    max_diff_count: int
    last_24h_comparisons: int
    last_7d_comparisons: int
